velox/connectors/WriteProtocol.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/WriteProtocol.h"
+#include "velox/connectors/Connector.h"
+
+#include <unordered_map>
+
+namespace facebook::velox::connector {
+
+namespace {
+
+using RegisteredWriteProtocols = std::unordered_map<
+    CommitStrategy,
+    std::function<std::shared_ptr<WriteProtocol>()>>;
+
+RegisteredWriteProtocols& registeredWriteProtocols() {
+  static RegisteredWriteProtocols protocols;
+  return protocols;
+}
+
+} // namespace
+
+// static
+void WriteProtocol::registerWriteProtocol(
+    CommitStrategy commitStrategy,
+    const std::function<std::shared_ptr<WriteProtocol>()> writeProtocol) {
+  bool addNew = registeredWriteProtocols()
+                    .insert_or_assign(commitStrategy, writeProtocol)
+                    .second;
+  if (!addNew) {
+    LOG(WARNING) << fmt::format(
+        "Write protocol for commit strategy {} is replaced", commitStrategy);
+  }
+}
+
+// static
+std::shared_ptr<WriteProtocol> WriteProtocol::newWriteProtocol(
+    CommitStrategy commitStrategy) {
+  const auto iter = registeredWriteProtocols().find(commitStrategy);
+  VELOX_CHECK(
```

Comment:
Would you document this behavior, e.g. failure in case write protocol for the specified commit strategy is not registered?